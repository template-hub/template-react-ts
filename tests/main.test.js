const path = require('path')
const sao = require('sao')

const generator = path.join(__dirname, '..')

describe('test', () => {
  it('defaults', async () => {
    const mockPromptAnswers = {
      name: 'cloud',
      username: 'username',
      email: 'placholder@mymail.com'
    }

    const stream = await sao.mock({ generator }, mockPromptAnswers)

    expect(stream.fileList).toMatchSnapshot('Generated files')

    const pkg = JSON.parse(await stream.readFile('package.json'))
    expect(pkg.name).toBe('cloud')
    expect(pkg.author).toBe('username <placholder@mymail.com>')

    expect(await stream.readFile('README.md')).toMatchSnapshot('README.md')
    expect(await stream.readFile('.editorconfig')).toMatchSnapshot('.editorconfig')
  })
})
