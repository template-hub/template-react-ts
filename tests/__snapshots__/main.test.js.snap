// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test defaults: .editorconfig 1`] = `
"root = true

[*]
indent_style = space
indent_size = 2
end_of_line = lf
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true

[*.md]
trim_trailing_whitespace = false
"
`;

exports[`test defaults: Generated files 1`] = `
Array [
  ".editorconfig",
  ".eslintrc.json",
  ".gitignore",
  ".npmrc",
  "LICENSE",
  "README.md",
  "package.json",
]
`;

exports[`test defaults: README.md 1`] = `
"# cloud

## License

MIT &copy; username
"
`;
